profile/src/main/java/com/scottlogic/deg/profile/reader/MainConstraintReader.java

```diff
@@ -36,7 +36,6 @@ public MainConstraintReader(AtomicConstraintTypeReaderMap constraintReaderMap) {
         this.constraintReaderMap = constraintReaderMap;
     }
 
-    @Override
     public Constraint apply(
```

Comment:
Maybe rename this method to differentiate it from the apply method in the classes implementing the AtomicConstraintReader? This might make it a little easier to see what is going on with the many calls to apply below.